### ISSUE TITLE
Link to private content of public Space results in 404

### ISSUE THREAD

**Issue by luke-** [humhub/humhub#6002]
### Is your feature request related to a problem? Please describe.

If the link to a private content of a public Space is shared to a non-member (of the Space), that user will receive a 404 error message.

### Describe the solution you'd like

Instead, it would be better to display a page, with a text like:
"This content is only available to members of this Space. Click here to become/request membership."

--- 
For content of a Private Space the generic 404 page is correct, however we should improve the text a bit here.
e.g. The requested content cannot be displayed. Either it was deleted, you mistyped it or it is not available for you.

**Comment by yurabakhtin** [comment]
@luke- As I understand you tell about core content records `Poll`, but probably we should apply the same my new code `ContentContainerController->checkContentIsReadable($record)` for other external module records like `Calendar`, `Task` and etc.

PR https://github.com/humhub/humhub/pull/6005 :

I could not implement the link to request a membership because there the code `Html::encode()` before display a message, so it looks like this:

![html_encoded](https://user-images.githubusercontent.com/6995524/210209643-8fd4048d-faa7-4ba7-80a6-5355f9031f72.png)

I have decided to use the message:

![message_403](https://user-images.githubusercontent.com/6995524/210209673-f866bfba-ddd8-49d7-9f6f-b053586c5fa8.png)

for 404:

![message_404](https://user-images.githubusercontent.com/6995524/210209784-90a5e4bd-7bf6-417b-a5fc-ef8e4deb63aa.png)

**Comment by luke-** [comment]
@yurabakhtin In the first case (Private Content & Public Space): I would like instead of the default Error View, a page in Space layout (with header). Maybe @Eladnarlea can also design it a bit.

The second case (Link to private Space): Please leave it as it is and don't change anything here for now.

**Comment by luke-** [comment]
@Semir1212 You you please add a message here. Maybe with headline. Please note, it's not always possible to become/request membership. Maybe some illu from @Eladnarlea ?

**Comment by Semir1212** [comment]
@luke- 

**Access denied**

You do not have permission to access this content, as it is reserved for members of this Space. Please become a member or apply for membership. The available options for membership will depend on the Space's settings.

**Comment by Eladnarlea** [comment]
<img width="668" alt="Screen Shot 2023-03-23 at 1 02 53 PM" src="https://user-images.githubusercontent.com/105652258/227198146-70c42a83-33dc-47bd-b07c-ef44316aab42.png">

How do you like something like this? or do you want me to design it more like an info?

**Comment by Eladnarlea** [comment]
<img width="1131" alt="Screen Shot 2023-03-23 at 2 21 17 PM" src="https://user-images.githubusercontent.com/105652258/227217065-fb1cbcb1-2c47-44d9-9f26-d1c7ee462f2b.png">

**Comment by luke-** [comment]
@Eladnarlea Yes, looks fine, but we should display the error message on full with 
![image](https://user-images.githubusercontent.com/4736168/227275480-abb74eb5-8615-4efb-bd85-ca85033785b0.png)

**Comment by Eladnarlea** [comment]
@luke-  you mean without the white border?

**Comment by yurabakhtin** [comment]
@luke- @Eladnarlea https://github.com/humhub/humhub/pull/6005/commits/ba3b40352f47877fbd74ac120e495060f7954bf1:

![space_private](https://user-images.githubusercontent.com/6995524/227453628-dd944981-446a-4c7f-8792-dd8d7d618733.png)

@Semir1212 Please provide a text for Profile private content:

![user_private](https://user-images.githubusercontent.com/6995524/227453701-a9d69641-1efe-4a4a-a9ee-87a96e8e2db1.png)

**Comment by Eladnarlea** [comment]
@yurabakhtin can you please make the padding top&bottom the same (less than the padding-top in your version) and transform the headline to h2 instead of h1?

**Comment by yurabakhtin** [comment]
@Eladnarlea Commit https://github.com/humhub/humhub/pull/6005/commits/70f680e55afad09f809a46971cc61640511689b4:

![space_private_styles](https://user-images.githubusercontent.com/6995524/227483251-6abe08cb-1d7a-4868-8e07-4bff1e3de941.png)

**Comment by Eladnarlea** [comment]
Thanks! :)

**Comment by Semir1212** [comment]
@yurabakhtin Text for profile posts:

Access denied

You do not have permission to access this content. The user has marked it as private.

**Comment by yurabakhtin** [comment]
@Semir1212 Commit https://github.com/humhub/humhub/pull/6005/commits/0aa682f7c4864485148a808e52f9c63ad590fba6:

![user_access_denied](https://user-images.githubusercontent.com/6995524/227863775-202cc3fb-6509-4304-9452-a7802881439c.png)
